src/posts/2020-09-19-how-to-write-accessible-forms.md

```diff
@@ -0,0 +1,158 @@
+```html
+<form>
+  <label for="firstName">First Name</label>
+  <input type="text" id="firstName" name="firtName"/>
+</form>
+```
+
+- Nest `<input>` directly inside `<label>`.
+```html
+<form>
+  <label>
+    First Name
+    <input type="text" name="firstName">
+  </label>
+</form>
+```
+
+**Note**: The first method is the preffered approach to associate a label with a form control . Additionally, some assistive devices (such as [Dragon NaturallySpeaking](https://en.wikipedia.org/wiki/Dragon_NaturallySpeaking)) do not support the latter.
+
+
+## Highlight input element on focus
+
+Form insertion [carets](https://developer.mozilla.org/en-US/docs/Glossary/caret#:~:text=A%20caret%20sometimes%20called%20a,is%20called%20the%20insertion%20point.) can be difficult to track. A simple solution is to highlight `<input>` on [focus](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus). This way, people do not have to guess where they are on the form. 
+
+![A form with an input field highlighted in blue.](../img/posts/2020-09-19-how-to-write-accessible-forms/form-field-on-focus.png)
+
+```css
+input:focus{
+  outline: 2px solid royalblue;
+  box-shadow: 1px 1px 8px 1px royalblue;
+}
+```
+
+## Break long forms into smaller sections
+
+Filling long forms can be stressful. This is largely true for almost everyone but particularity for people  with [AD/HD](https://www.cdc.gov/ncbddd/adhd/facts.html). Such forms can lead to an increase in [Bounce rate](https://en.wikipedia.org/wiki/Bounce_rate) which isn't what anyone wants. A simple solution is to break them into smaller sections. 
+
+This helps make long forms less daunting and easier to understand as it is provided in bits.
+
+
+The following principles apply for multi-step forms:
+- Repeat overall instruction on every page.
+- The forms should be split into logical groups.
+- The forms should inform the person about the progress thay are making.
+
+
+## Provide error messages if any
+
+If your site has forms, validation messaging and errors are inevitable.
+
+When people encounter errors, clearly they need more help. Some people, like the visually impaired, may not even be aware that an error has occurred. It is common practice to change `<input>` color to red which is not ideal. This is because assistive devices will not catch the error, hence not informing the person of the error.
+
+The following are ways to effectively and accessibly communicate form errors:
+- Let the person know there is an error.
+- Let the person access the error, and provide instructions on how to fix it.
+- Allow the person re-submit the form.
+
+
+In the code below, there is an empty [live region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) which report errors to assistive technologies. Additionally, each field is followed by a _hint_ which is hidden by default but shown if that field does not validate.

```

Comment:
@scottaohara What's your take on a live region for error reporting versus updating the `title` element and an input error summary section?